DataCapturing/Source/Synchronization/Upload/PreRequest.swift

```diff
@@ -80,13 +80,16 @@ class PreRequest {
                 }
 
                 let status = response.statusCode
-                guard let location = response.headers["Location"] else {
-                    onFailure(measurementIdentifier, ServerConnectionError.noLocation)
-                    return
-                }
 
                 if status == 200 {
+                    guard let location = response.headers["Location"] else {
+                        onFailure(measurementIdentifier, ServerConnectionError.noLocation)
+                        return
+                    }
+
                     onSuccess(authToken, location, upload)
+                } else if status == 412 {
+                    onFailure(measurementIdentifier, ServerConnectionError.uploadNotAccepted(measurementIdentifier: Int64(measurementIdentifier)))
```

Comment:
As I was not sure what to do with this case the app currently only shows an error symbol on such occasions. The user can delete the measurement herself. But I can create a task to delete such measurements (iOS does delete uploaded measurements completely, so that should probably happen here too).

Reply:
It could be weird for the user when a measurement is collected (without the knowledge of the user the geo data is missing), then the background upload starts and deletes the measurement as the server is not interested in measurements without locations.
I.e. next time the user searches for the measurement (when he notices it's not uploaded) he won't find a trace about what went wrong.

That's why I have the specific state "skipped" on Android, so the user can see that for whatever reasons, the measurement was not uploaded, and the user can contact us with the measurement id still visible, for further explanation, if necessary. (or the UI could show information in which cases uploads are skipped)

But I guess there are more important tasks right now, so better keep the measurement (and just don't retry uploading) than just deleting it without any trace whatsoever I think.

This way the user can at least see, that the upload failed and see that the length of the measurement was 0.0.

Reply:
You might, however, want to add documentation in the code that such measurements are not deleted on purpose, so that the user can see that there was an issue with the measurement.